File: sunghyun/PGS/1028/카펫.java
import java.util.*;

class Solution {
    public int[] solution(int brown, int yellow) {
        int sum = (brown - 4) / 2;
        int h = 0;
        for (int i = 1; i <= sum / 2; i++){
            if (i * (sum - i) == yellow) {
                h = i;
                break;
            }
        }
        int[] wh = {sum - h + 2, h + 2};
        return wh;
    }
}

File: dayeong/PGS/완전탐색/카펫.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int brown, int yellow) {
    vector<int> answer;
    int tmp = brown / 2 - 2;
    int height = 0;
    int width = 0;
    
    while(width >= height){
        height++;
        
        width = tmp - height;
        
        if(height * width == yellow){
            answer.push_back(width + 2);
            answer.push_back(height + 2);
            break;
        }
    }
    
    return answer;
}